src/Databases/DatabaseAtomic.cpp

```diff
@@ -180,6 +180,55 @@ void DatabaseAtomic::dropTableImpl(ContextPtr local_context, const String & tabl
     DatabaseCatalog::instance().enqueueDroppedTableCleanup(table->getStorageID(), table, table_metadata_path_drop, sync);
 }
 
+void DatabaseAtomic::dropDetachedTable(ContextPtr local_context, const String & table_name, const bool sync)
+{
+    waitDatabaseStarted();
+
+    assert(fs::exists(getObjectMetadataPath(table_name)));
+
+    const String table_metadata_path = getObjectMetadataPath(table_name);
+    const UUID uuid_table = getTableUUIDFromDetachedMetadata(local_context, table_metadata_path);
+    const StorageID storage_id{getDatabaseName(), table_name, uuid_table};
+    String table_metadata_path_drop;
+
+    {
+        std::lock_guard lock(mutex);
+        table_metadata_path_drop = DatabaseCatalog::instance().getPathForDroppedMetadata(storage_id);
+
+        fs::create_directory(fs::path(table_metadata_path_drop).parent_path());
+
+        auto txn = local_context->getZooKeeperMetadataTransaction();
+        if (txn && !local_context->isInternalSubquery())
+            txn->commit();
+
+        fs::rename(table_metadata_path, table_metadata_path_drop);
+        LOG_TRACE(log, "Rename {} to {} for removing.", table_metadata_path, table_metadata_path_drop);
+
+        table_name_to_path.erase(table_name);
+    }
+
+    if (fs::exists(getPathSymlink(table_name)))
+    {
+        LOG_TRACE(log, "Remove symlink for {}", table_name);
+        tryRemoveSymlink(table_name);
+    }
+
+    {
+        std::lock_guard lock(mutex);
+        detached_tables.erase(uuid_table);
```

Comment:
This breaks `DatabaseAtomic::waitDetachedTableNotInUse`: it still may be in use after this, but `waitDetachedTableNotInUse` will return

Reply:
Problem was fixed